### PR TITLE
[EXE-1951] [Spark 2] Implement aiq_day_diff pushdown

### PR DESCRIPTION
**Pull request by dorisZ017**
Implement BigQuery SQL equivalent of aiq_day_diff; 
Adding unit test and integration test;
Fixing some issues with integration test; adding to README

Installed locally and pulled into flame; tested with spark-shell in flame

### CODE REVIEW

File: spark-bigquery-connector-common/src/test/java/com/google/cloud/spark/bigquery/integration/SparkBigQueryIntegrationTestBase.java [hunk 1]
```diff
@@ -44,12 +44,11 @@ protected static class SparkFactory extends ExternalResource {
...
       spark =
           SparkSession.builder()
-              .master("local")
+              .master("local[*]")
               .config("spark.ui.enabled", "false")
               .config("spark.default.parallelism", 20)
               .getOrCreate();
-      // reducing test's logs
-      spark.sparkContext().setLogLevel("WARN");
+      spark.sparkContext().setLogLevel("INFO"); // helpful for viewing queries running in BQ
```
Comment (ShaoFuWu) [inline spark-bigquery-connector-common/src/test/java/com/google/cloud/spark/bigquery/integration/SparkBigQueryIntegrationTestBase.java:51]: just curious, can we override this in runtime, when running the test?

File: README.md [hunk 2]
```diff
@@ -1249,14 +1249,45 @@ This places artifacts in `~/.m2/repository/`
...
-# Tests
+# Unit Tests
 Full tests dont work, they require configuring GCP settings, so just make sure projects
 we care about are passing. Or if you are making a change, confirm the same tests fail with
 and without the change.
 ```
 ./mvnw test -fn
 ```
 
+# Integration Tests
```
Comment (ShaoFuWu) [inline README.md:1260]: thanks for writing readme 👍

File: spark-bigquery-pushdown/pushdown_common_src/main/scala/com/google/cloud/spark/bigquery/pushdowns/SparkExpressionConverter.scala [hunk 3]
```diff
@@ -215,6 +215,20 @@ abstract class SparkExpressionConverter {
...
             convertStatement(date, fields) + s", ${format.toString()}"
           )
+      /**
+       * aiq_day_diff(startTsExpr, endTsExpr, timezoneExpr) =>
+       * DATE_DIFF(
+       *   DATE(TIMESTAMP_MILLIS(endTsExpr), timezoneExpr),
+       *   DATE(TIMESTAMP_MILLIS(startTsExpr), timezoneExpr),
+       *   DAY
+       * )
+       */
```
Comment (ShaoFuWu) [inline spark-bigquery-pushdown/pushdown_common_src/main/scala/com/google/cloud/spark/bigquery/pushdowns/SparkExpressionConverter.scala:225]: 💯

**Review by ShaoFuWu — APPROVED** [review]

File: spark-bigquery-connector-common/src/test/java/com/google/cloud/spark/bigquery/integration/ReadIntegrationTestBase.java [hunk 4]
```diff
@@ -429,6 +429,7 @@ public void testReadFromBigLakeTable_parquet() {
   }
 
   @Test
+  @Ignore("aiq")
```
Comment (MasterDDT) [inline spark-bigquery-connector-common/src/test/java/com/google/cloud/spark/bigquery/integration/ReadIntegrationTestBase.java:432]: Curious, why does this one fail?

File: spark-bigquery-connector-common/src/test/java/com/google/cloud/spark/bigquery/integration/TestConstants.java [hunk 5]
```diff
@@ -73,15 +72,8 @@ public class TestConstants {
...
-      Preconditions.checkNotNull(
-          System.getenv(TEMPORARY_GCS_BUCKET_ENV_VARIABLE),
-          "Please set the %s env variable to point to a write enabled GCS bucket",
-          TEMPORARY_GCS_BUCKET_ENV_VARIABLE);
-  static final String BIGLAKE_CONNECTION_ID =
-      Preconditions.checkNotNull(
-          System.getenv(BIGLAKE_CONNECTION_ID_ENV_VARIABLE),
-          "Please set the BIGLAKE_CONNECTION_ID env variable in order to create biglake table");
+  static final String TEMPORARY_GCS_BUCKET = "";
+  static final String BIGLAKE_CONNECTION_ID = "";
```
Comment (MasterDDT) [inline spark-bigquery-connector-common/src/test/java/com/google/cloud/spark/bigquery/integration/TestConstants.java:76]: I doubt we will want to run it, but maybe better to keep the code to read from env value, and make the default value empty string

File: spark-bigquery-pushdown/pushdown_common_src/test/scala/com/google/cloud/spark/bigquery/pushdowns/SparkExpressionConverterSuite.scala [hunk 6]
```diff
@@ -943,6 +944,15 @@ class SparkExpressionConverterSuite extends AnyFunSuite with BeforeAndAfter {
     assert(bigQuerySQLStatement.get.toString == "DATE_TRUNC ( '2016-07-30' , YEAR )")
   }
 
+  test("convertDateExpressions with AiqDayDiff") {
+    val dayDiff = AiqDayDiff(startMsAttributeReference, Literal(1695945601000L), Literal("UTC"))
```
Comment (MasterDDT) [inline spark-bigquery-pushdown/pushdown_common_src/test/scala/com/google/cloud/spark/bigquery/pushdowns/SparkExpressionConverterSuite.scala:948]: Want to also test the return value is correct?

File: spark-bigquery-connector-common/src/test/java/com/google/cloud/spark/bigquery/integration/SparkBigQueryIntegrationTestBase.java (hunk 1 above)
Comment (dorisZ017) [reply spark-bigquery-connector-common/src/test/java/com/google/cloud/spark/bigquery/integration/SparkBigQueryIntegrationTestBase.java:51]: Hmm looks like it's controlled by some log4j config, not sure if there's an easy way to override from mvn commands

File: spark-bigquery-pushdown/pushdown_common_src/test/scala/com/google/cloud/spark/bigquery/pushdowns/SparkExpressionConverterSuite.scala (hunk 6 above)
Comment (dorisZ017) [reply spark-bigquery-pushdown/pushdown_common_src/test/scala/com/google/cloud/spark/bigquery/pushdowns/SparkExpressionConverterSuite.scala:948]: yeah unit tests only check if the queries are correct, there are integration tests that read from a real bigquery table and check the query results

**Comment by dorisZ017** [comment]
Published artifacts
```
s3://aiq-artifacts/releases/com/google/cloud/spark/spark-2.4-bigquery/0.28.0-aiq12/
```

**Pull request merged**